sparksteps/steps.py

```diff
@@ -29,6 +30,15 @@ def zip_to_s3(s3_resource, dirpath, bucket, key):
     return response
 
 
+def parse_s3_path(s3_path):
+    """Returns bucket, path, and filename of an S3 path"""
+    parsed = urlparse(s3_path, allow_fragments=False)
+    bucket = parsed.netloc
+    path, filename = parsed.path.rsplit('/', 1)
+    path = path[1:] if path.startswith('/') else path
```

Comment:
This is necessary to avoid problems with os.path.join:

```
>>> os.path.join('foo', '/bar') 
'/bar'
>>> os.path.join('foo', 'bar') 
'foo/bar'
```